File: BACKEND/test_flask.py
import requests
import base64
import os

def test_flask():
    image_path = 'test_image.png'
    if not os.path.exists(image_path):
        print("No test image found.")
        return

    with open(image_path, 'rb') as f:
        img_base64 = base64.b64encode(f.read()).decode('utf-8')

    try:
        response = requests.post('http://localhost:5001/api/analyze-craft', json={'image': img_base64})
        print("Flask Response:", response.json())
    except Exception as e:
        print("Error:", e)

if __name__ == "__main__":
    test_flask()

File: BACKEND/test_flask.js
const fetch = require('node-fetch');
const fs = require('fs');

async function testFlask() {
  try {
    const imagePath = './test_image.png'; // Using existing test image in BACKEND
    if (!fs.existsSync(imagePath)) {
        console.log("No test image found.");
        return;
    }
    const imageBase64 = fs.readFileSync(imagePath, { encoding: 'base64' });

    const response = await fetch('http://localhost:5001/api/analyze-craft', {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({ image: imageBase64 }),
    });

    const data = await response.json();
    console.log("Flask Response:", JSON.stringify(data, null, 2));
  } catch (error) {
    console.error("Test Error:", error);
  }
}

testFlask();
